### DISCUSSION TITLE
How do i remove the white background from ftxui::Input elements?

### DISCUSSION THREAD

**Discussion by cedogibi** [ArthurSonzogni/FTXUI#960] [Q&A] [answered]
I have recently been getting trouble with trying to remove this white background from the ftxui::Input element.
It makes the text barely readably and overall its not very nice.
Code: 
```cpp
 std::string inputValue;
  auto input = ftxui::Input(&inputValue, "Enter input");

  auto inputWithEnterHandler = ftxui::CatchEvent(input, [&](ftxui::Event event) {
    if (event == ftxui::Event::Return) {
      inputValue.clear();
      return true;
    }
    return false;
  });

  auto inputContainer = ftxui::Renderer(inputWithEnterHandler, [&] {
    return ftxui::vbox(
               {ftxui::text("Example") | ftxui::bold | ftxui::center, ftxui::filler(),
                inputWithEnterHandler->Render() | ftxui::color(ftxui::Color::GrayLight) |
                    ftxui::border}
           ) |
           ftxui::size(ftxui::HEIGHT, ftxui::LESS_THAN, 24) | ftxui::border |
           ftxui::bgcolor(ftxui::Color::RGB(20, 20, 20));
  });

  auto screen = ftxui::ScreenInteractive::TerminalOutput();
  screen.Loop(inputContainer);
```
Output:
![Screenshot 2024-11-29 20 47 30](https://github.com/user-attachments/assets/cdb10d81-a3e6-4590-a604-14ce7f34a618)

**Comment by ArthurSonzogni** [comment, marked as answer]
Hello @a-cream 

Did you try to provide an `InputOption::tranform` to `Input`? This should give you the freedom to customize it.

See [demo](https://arthursonzogni.github.io/FTXUI/examples/?file=component/input_style) from [examples/component/input_style.cpp](https://github.com/ArthurSonzogni/FTXUI/blob/main/examples/component/input_style.cpp)

**Reply by cedogibi** [reply]
Thank you, your library is so awesome!
